backend/src/database/image_controller.rs

```diff
@@ -1,35 +1,85 @@
 use crate::general_helpers::ENV_VARS;
+use uuid::Uuid;
+use sqlx::{postgres::{PgPoolOptions, PgQueryResult}, QueryBuilder, Pool, Postgres, Row};
 
-use super::types::image::Image;
+use super::{
+    types::{image::Image, post::Post},
+    helpers::{DATABASE_URL, ensure_tables_exist},
+};
 
 fn get_mock_data() -> Vec<Image> {
     return vec![
         Image {
             id: "1".to_string(),
-            timestamp: 0,
             data: vec![0, 1, 2],
+            timestamp: 0,
         },
         Image {
             id: "2".to_string(),
-            timestamp: 1,
             data: vec![0, 5, 10],
+            timestamp: 1,
         },
         Image {
             id: "3".to_string(),
-            timestamp: 2,
             data: vec![10, 20, 30],
+            timestamp: 2,
         },
     ];
 }
 
-pub fn read_images(ids: Vec<String>) -> Vec<Image> {
+pub async fn read_images(ids: Vec<String>) -> Result<Vec<Image>, sqlx::Error> {
     if ENV_VARS.use_mocked_database {
-        return get_mock_data()
+        return Ok(get_mock_data()
             .iter()
             .filter(|image| ids.contains(&image.id))
             .map(|image| image.clone())
-            .collect::<Vec<Image>>();
+            .collect::<Vec<Image>>());
     } else {
-        todo!("Implement this part of the database interaction");
+        return Ok(read_images_from_db(ids).await?);
     }
 }
+
+async fn read_images_from_db(ids: Vec<String>) -> Result<Vec<Image>, sqlx::Error> {
+    let pool = PgPoolOptions::new().max_connections(5).connect(DATABASE_URL).await?;
+
+    let mut querybuilder_stub = QueryBuilder::new("select * from images where ");
+    let finalized_querybuilder = querybuilder_stub.push(ids.join(" or "));
```

Comment:
Er ikke lige inde i den her querybuilder ting, men vil disse to linjer ikke resultere i `"select * from images where id1 or id2 or id3"`

Reply:
det er en gyldig query, men der mangler et semikolon 🥲

Reply:
men skal du ikke definere hvilket felt der er tale om?